### PR TITLE
feat: UX improvements to Action

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Ended up doing some refactoring in [this PR](https://github.com/PostHog/posthog/pull/22056) so decided to pull it out

## Changes

* Proper loading and not found state for actions
* Smaller action segmented buttons
* Tighter padding on match groups
* Clearer secondary button for adding another match group
* Fixed unloading logic
* Container based re-flowing of the grid (works better with the sidebar)
* Redirect back to the action when undoing delete
* Disable the save button when not changed and hide it when editing unless changed (makes it more obvious how to save it when you make a change)
* Set the default for an match group to be a $pageview
* Re-order options to be `Pageview, Autocapture, Other events` as that felt more readable

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Review by pauldambra — APPROVED** [review]
didn't run it but lgtm

**Comment by pauldambra** [comment]
i guess we have no visual regression tests for this scene 🙈

**Review by pauldambra — APPROVED** [review]
ran it, looks great...

cypress failure looks real (probably cypress test too tightly bound to vague selector from the look of it)

File: frontend/src/scenes/actions/ActionEdit.tsx [hunk 1]
```diff
@@ -136,20 +135,23 @@ export function ActionEdit({ action: loadedAction, id }: ActionEditLogicProps):
...
+                                >
+                                    Save
+                                </LemonButton>
+                            ) : null}
                         </>
                     }
                 />
 
-                <div>
+                <div className="@container">
```
Comment (daibhin) [inline frontend/src/scenes/actions/ActionEdit.tsx:154]: What do the `@` symbols do in these cases?

**Review by daibhin — APPROVED** [review]
Some screenshots of the changes or snapshot stories would be nice to see what's changing here

File: frontend/src/scenes/actions/ActionEdit.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/scenes/actions/ActionEdit.tsx:154]: This is the tailwind container queries format.

**Pull request merged**